src/magic_folder/cli.py

```diff
@@ -333,126 +328,151 @@ def parseArgs(self, alias, nickname=None, local_dir=None):
                 "--poll-interval must be a positive integer"
             )
 
-        # Expand the path relative to the current directory of the CLI command, not the node.
-        self.local_dir = None if local_dir is None else argv_to_abspath(local_dir, long_path=False)
-
-        if self.nickname and not self.local_dir:
-            raise usage.UsageError("If NICKNAME is specified then LOCAL_DIR must also be specified.")
-
 
 @inlineCallbacks
-def create(options):
-    precondition(isinstance(options.alias, unicode), alias=options.alias)
-    precondition(isinstance(options.nickname, (unicode, NoneType)), nickname=options.nickname)
-    precondition(isinstance(options.local_dir, (unicode, NoneType)), local_dir=options.local_dir)
+def add(options):
+    """
+    Add a new Magic Folder
+    """
 
-    try:
-        from twisted.internet import reactor
-        treq = HTTPClient(Agent(reactor))
-
-        name = options['name']
-        nodedir = options.parent.node_directory
-        localdir = options.local_dir
-        rc = yield _create(options.alias, options.nickname, name, nodedir, localdir, options["poll-interval"], treq)
-        print("Alias %s created" % (quote_output(options.alias),), file=options.stdout)
-    except Exception as e:
-        print("%s" % str(e), file=options.stderr)
-        returnValue(1)
+    from twisted.internet import reactor
+    treq = HTTPClient(Agent(reactor))
+    yield magic_folder_create(
+        options.parent.config,
+        argv_to_unicode(options["name"]),
+        argv_to_unicode(options["author"]),
+        options.local_dir,
+        options["poll-interval"],
+        treq,
+    )
+    print("Created magic-folder named '{}'".format(options["name"]), file=options.stdout)
```

Comment:
The `yield` here actually gets used/abused by the `@inlineCallbacks` stuff so that it works "kind of like" an `await` would. So, we're awaiting the result of the Deferred from `magic_folder_create`. The inlineCallbacks makes this function return a Deferred .. and also makes that Deferred "failed" if the one we're waiting for fails.

So, the user will see a traceback / failed Deferred **or** execution continues and the `print` runs.